ament_uncrustify/ament_uncrustify/main.py

```diff
@@ -301,12 +226,118 @@ def get_files(paths, extensions, excludes=[]):
                     if filename in excludes:
                         continue
                     _, ext = os.path.splitext(filename)
-                    if ext not in ['.%s' % e for e in extensions]:
-                        continue
-                    files.append(os.path.join(dirpath, filename))
+                    language = extensions_with_dot_to_language.get(ext, None)
+                    if language is not None:
+                        files[language].append(
+                            os.path.normpath(os.path.join(dirpath, filename)))
         if os.path.isfile(path):
-            files.append(path)
-    return [os.path.normpath(f) for f in files]
+            _, ext = os.path.splitext(path)
+            language = extensions_with_dot_to_language.get(ext, None)
+            files[language].append(os.path.normpath(path))
```

Comment:
@dirk-thomas consider checking if `language is not None` here too, or drop the default value in `get()` above.

Reply:
Why should this check for `language is not None`? We need the `None` key to enumerate all files with unknown extension.

Reply:
Oh, ok, to deal with the explicitly given files with unknown extension. Understood. Disregard.